flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -549,7 +559,12 @@ protected void runWithArgs(String[] args, String terminateAfterString, String[]
 		final int startTimeoutSeconds = 180;
 		final long deadline = System.currentTimeMillis() + (startTimeoutSeconds * 1000);
 
-		Runner runner = new Runner(args, type, expectedReturnValue);
+		Runner runner = new Runner(
+			args,
+			flinkConfiguration,
+			CliFrontend.getConfigurationDirectoryFromEnv(),
```

Comment:
Since the configuration directory is only required for the session case in `Runner`, the same argument applies as above. I agree that the `FlinkYarnSessionCli` should have a configurable configuration directory which can be controlled by a command line option.